src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferConstCharOverConstUnitStringAnalyzer.cs

```diff
@@ -0,0 +1,143 @@
+                if (!compilationContext.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemTextStringBuilder, out INamedTypeSymbol? stringBuilderType))
+                {
+                    return;
+                }
+
+                IMethodSymbol appendStringMethod = stringBuilderType
+                    .GetMembers("Append")
+                    .OfType<IMethodSymbol>()
+                    .WhereAsArray(s =>
+                        s.Parameters.Length == 1 &&
+                        s.Parameters[0].Type.SpecialType == SpecialType.System_String)
+                    .FirstOrDefault();
+                if (appendStringMethod is null)
+                {
+                    return;
+                }
+
+                compilationContext.RegisterOperationAction(operationContext =>
+                {
+                    IInvocationOperation invocationOperation = (IInvocationOperation)operationContext.Operation;
+                    if (invocationOperation.Arguments.Length < 1)
+                    {
+                        return;
+                    }
+
+
+                    if (!invocationOperation.TargetMethod.Equals(appendStringMethod))
+                    {
+                        return;
+                    }
+
+                    ImmutableArray<IArgumentOperation> arguments = invocationOperation.Arguments;
+                    IArgumentOperation firstArgument = arguments[0];
+
+                    // We don't handle class fields. Only local declarations
+                    IOperation firstArgumentValue = firstArgument.Value;
+                    if (firstArgumentValue is ILocalReferenceOperation argumentValue)
+                    {
+                        HandleLocalReferenceOperation(argumentValue, operationContext);
+                        return;
+                    }
+                    else if (firstArgumentValue is ILiteralOperation literalArgumentValue)
+                    {
+                        HandleLiteralReferenceOpearion(literalArgumentValue, operationContext);
+                        return;
+                    }
+                    return;
+                }
+                , OperationKind.Invocation);
+            });
+        }
+
+        private static void HandleLiteralReferenceOpearion(ILiteralOperation literalArgumentValue, OperationAnalysisContext operationContext)
+        {
+            if (literalArgumentValue.ConstantValue.HasValue && literalArgumentValue.ConstantValue.Value is string unitString && unitString.Length == 1)
+            {
+                operationContext.ReportDiagnostic(literalArgumentValue.CreateDiagnostic(Rule));
+            }
+        }
+
+        private static void HandleLocalReferenceOperation(ILocalReferenceOperation argumentValue, OperationAnalysisContext operationContext)
+        {
+            ILocalSymbol localArgumentDeclaration = argumentValue.Local;
+
+            // If there are multiple declarations, bail
```

Comment:
Why are we doing this bailout? I think the entire code block from line 107 to 134 can we removed.

Reply:
Okay, so looking at the analyzer and code fix code it seems you have multiple declarations bailout as you want to avoid dealing with the complications of updating that code on the code fixer site. It is fine to not handle this case in the code fix, but I think you should still be flagging this case in the analyzer. I would recommend removing line 107 to 134 completely from the analyzer, but still have the bail out code in the code fixer - for multideclaration case, user has to fix his code manually.

Reply:
Ok, fixed.